src/retrace/retrace.py

```diff
@@ -357,79 +355,127 @@ def run_gdb(savedir, plugin):
+                          "--", "/bin/chmod a+rx /var/spool/abrt/gdb.sh"],
+                         stdout=null, stderr=null)
+            if chmod:
+                raise Exception("Unable to chmod GDB launcher")
+
+            child = Popen(["/usr/bin/mock", "shell", "--configdir", savedir,
+                           "--", "su mockbuild -c '/bin/sh /var/spool/abrt/gdb.sh'",
+                           # redirect GDB's stderr, ignore mock's stderr
+                           "2>&1"], stdout=PIPE, stderr=null, encoding='utf-8')
+
+        elif CONFIG["RetraceEnvironment"] == "podman":
+            podman_build_call = ["/usr/bin/podman", "build", "--file",
+                                 os.path.join(savedir, RetraceTask.DOCKERFILE),
+                                 "--volume=%s:%s:ro" % (repopath, repopath)]
+            if CONFIG["UseFafPackages"]:
+                log_debug("Using FAF repository '%s'" % fafrepo)
+                podman_build_call.append("--volume=%s:/var/spool/abrt/faf-packages" % fafrepo)
+            podman_build_call.extend(["--tag", "retrace-image"])
+
+            build_image = call(podman_build_call, stdout=null, stderr=null)
+            if build_image:
+                raise Exception("Unable to build podman container")
+
+            if CONFIG["UseFafPackages"]:
+                child = Popen(["/usr/bin/podman",
+                               "run",
+                               "-it",
+                               "--detach",
+                               "--volume=%s:%s:ro" % (repopath, repopath),
+                               "--volume=%s:/var/spool/abrt/faf-packages" % fafrepo,
+                               "retrace-image"],
+                              stdout=PIPE, stderr=PIPE, encoding='utf-8')
+                container_id = child.communicate()[0]
+                container_id = container_id.rstrip()
+
+                log_debug("Retrace container id: %s" % container_id)
+
+                child = Popen("/usr/bin/podman exec -it %s bash -c 'for PKG in /var/spool/abrt/faf-packages/*; do rpm2cpio $PKG | cpio -muid --quiet; done'" % container_id, shell=True, stdout=null, stderr=null, encoding='utf-8')
+                if child.wait():
+                    raise Exception("Unpacking of packages failed")
+                child = Popen(["/usr/bin/podman", "exec", container_id,
+                               "/var/spool/abrt/gdb.sh"], stdout=PIPE, stderr=PIPE, encoding='utf-8')
+            else:
+                child = Popen(["/usr/bin/podman", "run", "-it", "--volume=%s:%s:ro" % (repopath, repopath),
+                               "retrace-image"], stdout=PIPE, stderr=PIPE, encoding='utf-8')
+        else:
+            raise Exception("RetraceEnvironment set to invalid value")
 
     backtrace = child.communicate()[0].strip()
     if child.wait():
         raise Exception("Running GDB failed")
 
+    if container_id:
+        with open(os.devnull, "w") as null:
```

Comment:
Could you use [DEVNULL](uhttps://docs.python.org/3/library/subprocess.html#subprocess.DEVNULLrl) instead?

Reply:
Sure, replaced in a new commit.